uint32 NodeId
uint32[]  i_feature
